gramps/plugins/gramplet/agestats.py

```diff
@@ -30,273 +31,264 @@
 #-------------------------------------------------------------------------
 from collections import defaultdict
 
+#-------------------------------------------------------------------------
+#
+# Gtk modules
+#
+#-------------------------------------------------------------------------
+from gi.repository import Gtk
+
 #------------------------------------------------------------------------
 #
 # Gramps modules
 #
 #------------------------------------------------------------------------
 from gramps.gen.plug import Gramplet
 from gramps.gen.lib import ChildRefType
+from gramps.gui.widgets import Histogram
+from gramps.gui.plug.quick import run_quick_report_by_name
 from gramps.gen.const import GRAMPS_LOCALE as glocale
 _ = glocale.translation.gettext
 
 class AgeStatsGramplet(Gramplet):
 
     def init(self):
+        self.gui.WIDGET = self.build_gui()
+        self.gui.get_container_widget().remove(self.gui.textview)
+        self.gui.get_container_widget().add(self.gui.WIDGET)
+        self.gui.WIDGET.show()
+
         self.max_age = 110
-        self.max_mother_diff = 40
-        self.max_father_diff = 60
-        self.chart_width = 60
+        self.max_mother_diff = 55
+        self.max_father_diff = 70
+
+    def build_gui(self):
+        top = Gtk.ScrolledWindow()
+        top.set_policy(Gtk.PolicyType.AUTOMATIC, Gtk.PolicyType.AUTOMATIC)
```

Comment:
You are putting a ScrolledWindow inside an already existing ScrolledWindow.  I suspect that this is causing the issue with redraw when first adding the Gramplet.  In any event when I removed that it seems to display properly.
<pre>
    def build_gui(self):
#         top = Gtk.ScrolledWindow()
#         top.set_policy(Gtk.PolicyType.AUTOMATIC, Gtk.PolicyType.AUTOMATIC)
        self.vbox = Gtk.Box(orientation=Gtk.Orientation.VERTICAL)
        self.vbox.set_margin_start(6)
        self.vbox.set_margin_end(6)
        self.vbox.set_margin_top(6)
        self.vbox.set_margin_bottom(6)
        self.vbox.set_spacing(12)
#         top.add(self.vbox)
        return self.vbox  #top
</pre>